bitnami/discourse/templates/_helpers.tpl

```diff
@@ -120,7 +120,29 @@ imagePullSecrets:
 {{- end -}}
 
 {{/*
-Get the user defined LoadBalancerIP for this release.
+Return the Discourse secret name
+*/}}
+{{- define "discourse.discourseSecretName" -}}
+{{- if not (empty .Values.discourse.existingSecret) }}
+    {{- .Values.discourse.existingSecret }}
+{{- else -}}
+    {{- printf "%s" (include "discourse.fullname" .) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if a secret object should be created
+*/}}
+{{- define "discourse.createSecret" -}}
```

Comment:
I think the best approach for this is to split the secrets into three, one for each of the main applications in this huge Chart. That would result in a simpler management and a more neat result IMHO.

For example, you can take a look at how EJBCA Chart addresses this problem:

- [externaldb-secrets](https://github.com/bitnami/charts/blob/master/bitnami/ejbca/templates/externaldb-secrets.yaml) : Used to store externaldb credentials
- [secrets](https://github.com/bitnami/charts/blob/master/bitnami/ejbca/templates/secrets.yaml): Used to store EJBCA's own sensitive data.

What's your opinion on this? If you think that is too much I can handle that myself 😀